ingestion/src/kldata.rs

```diff
@@ -106,66 +152,84 @@ fn parse_columns(cols_raw: &str) -> Result<Vec<ObsinnId>, Error> {
         .collect::<Result<Vec<ObsinnId>, Error>>()
 }
 
-fn parse_obs(csv_body: Lines<'_>, columns: &[ObsinnId]) -> Result<Vec<ObsinnObs>, Error> {
+fn parse_obs(
+    csv_body: Lines<'_>,
+    columns: &[ObsinnId],
+    params: Arc<HashMap<String, Param>>,
```

Comment:
I think we should give this argument a better name to make clear that it doesn't refer to the params in the message (which are in fact encoded in `columns`) but rather our general reference list of params. Perhaps "params_reference" or "param_types"?